timelock-impl/src/main/java/com/palantir/atlasdb/backup/AtlasBackupResource.java

```diff
@@ -132,7 +132,7 @@ private CompletedBackup fetchFastForwardTimestamp(InProgressBackupToken backupTo
         long fastForwardTimestamp = timelock(namespace).getFreshTimestamp();
         return CompletedBackup.builder()
                 .namespace(namespace)
-                .backupStartTimestamp(backupToken.getBackupStartTimestamp())
+                .backupStartTimestamp(backupToken.getImmutableTimestamp())
```

Comment:
I don't know enough about the internals here to know if this is safe or not.

Reply:
yeah, we need to be careful about how to handle the gap between the immutable timestamp and backupStartTimestamp - my intuition was that we might not be able to rely on data in between immutableTs and backupTs, so we want to clear the transactions table for that range (so that we invalidate any data that may have been partially written at these timestamps at the time of backup).
If we invalidate all data from `(ImmutableTimestamp, BackupStarttimestamp]`, then we can probably restore back to `immutableTimestamp` (which I'm fairly sure is what happens currently in the backup/restore process).

Reply:
Anything that's after immutable timestamp and before the backup ts is fair game because we don't start the backup until after both timestamps are returned by timelock. In general, it's not clear to me why we have both and in particular seems unintuitive to restore at immutable timestamp than at backup timestamp

Reply:
what do you mean by "fair game" - I'm worried about this kind of sequence:

1. immutableTimestamp is T1
2. A transaction T starts at T2
3. Transaction T writes some data to table A (and sleeps)
4. Backup starts at T3
5. Backup backs up data from table A
6. Backup backs up data from table B 
7. Transaction T writes some data to table B (this is not present in the backup)
8. Transaction T commits at T4
9. Backup completes at T5
.....(time passes)...
10. We restore to the (T3, T5) backup. For the transaction at T2, we have data from table A, but no data from table B.
11. Severe data corruption

Is there anything to stop this sequence from happening, without cleaning the transaction table range from T1 to T3?

Reply:
Well that depends on the commit timestamp, we either have the entry in the transactions table in which case it is committed or not in which case all of it is dead, no?

Reply:
right, but the backup could be missing some parts of the commit (in particular due to the ordering of 6 and 7 above), so I wouldn't feel completely safe keeping the (immutableTs, backupTs] data here. @jeremyk-91 , thoughts?

Reply:
Looking at the docs for the clean transactions table task, I think we have correctness here, but we should record both the immutable and backupStart timestamps to retain current performance:
```
     * <p>This means aborting all transactions (1) whose commit timestamp is greater than the backup timestamp and (2)
     * whose start timestamp is greater than the immutable timestamp (that was locked during the backup).
     * <p>In practice there is no difference between the set of transactions that meet condition (1) and those that meet
     * both conditions (1) and (2). If the immutable timestamp is unknown this task will still be correct; being able to
     * apply condition (2) is simply a performance optimization.
```

So we should record both timestamps, and the CleanTransactionsTable phase should resolve the sequence that I was concerned about above.